pkg/lint/lintersdb/validator.go

```diff
@@ -14,9 +14,29 @@ type Validator struct {
 }
 
 func NewValidator(m *Manager) *Validator {
-	return &Validator{
-		m: m,
+	return &Validator{m: m}
+}
+
+// Validate validates the configuration by calling all other validators for different
+// sections in the configuration and then some additional linter validation functions.
+func (v Validator) Validate(cfg *config.Config) error {
+	err := cfg.Validate()
+	if err != nil {
+		return err
+	}
+
+	validators := []func(cfg *config.Linters) error{
+		v.validateLintersNames,
+		v.validatePresets,
```

Comment:
looks like `Validator` should make checks that requires knowledge from manager and config simultaneously.

if validation requires only on of them then need to move it to corresponding type (manager / config)

Reply:
The validator is only created inside the constructor of the manager, so it's already a "detail" of the manager.